Tests/TuistIntegrationTests/Generator/MultipleConfigurationsIntegrationTests.swift

```diff
@@ -250,6 +250,29 @@ final class MultipleConfigurationsIntegrationTests: XCTestCase {
         XCTAssertTrue(debug.contains("TARGET", "YES")) // from target settings
     }
 
+    func testGenerateWhenCustomDebugVariant() throws {
+        // Given
+        let projectDebugConfiguration = Configuration(settings: ["A": "A_PROJECT_DEBUG",
+                                                                 "B": "B_PROJECT_DEBUG"])
+        let projectReleaseConfiguration = Configuration(settings: ["A": "A_PROJECT_RELEASE",
```

Comment:
`projectReleaseConfiguration` here may be a bit misleading

Reply:
Updated the var name. For `SWIFT_COMPILATION_MODE` to work we will need to update `XcodeProj`. We're on a revision from 2nd Apr.

Happy to update XcodeProj to the latest `c17e1af8b806f16729da14263cf0435e9fbbd176` (in a separated RP). 
I've run the tests locally, all green. One new warning in XcodeProj:

```
XcodeProj/Extensions/String+md5.swift:28:32: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
            let message = data.withUnsafeBytes { bytes -> [UInt8] in
```